nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py

```diff
@@ -0,0 +1,272 @@
+
+            loss_mask.append(1)
+            loss_mask.extend([not ignore_extra_tokens] * (len(word_tokens) - 1))
+
+            subtokens_mask.append(1)
+            subtokens_mask.extend([0] * (len(word_tokens) - 1))
+
+            if with_label:
+                slots.extend([raw_slots[i][j]] * len(word_tokens))
+
+        subtokens.append(tokenizer.sep_token)
+        loss_mask.append(1 - ignore_start_end)
+        subtokens_mask.append(0)
+        sent_lengths.append(len(subtokens))
+        all_subtokens.append(subtokens)
+        all_loss_mask.append(loss_mask)
+        all_subtokens_mask.append(subtokens_mask)
+        all_input_mask.append([1] * len(subtokens))
+        if with_label:
+            slots.append(pad_label)
+            all_slots.append(slots)
+
+    max_seq_length = min(max_seq_length, max(sent_lengths))
+    logging.info(f'Max length: {max_seq_length}')
+    get_stats(sent_lengths)
+    too_long_count = 0
+
+    for i, subtokens in enumerate(all_subtokens):
+        if len(subtokens) > max_seq_length:
+            subtokens = [tokenizer.cls_token] + subtokens[-max_seq_length + 1 :]
+            all_input_mask[i] = [1] + all_input_mask[i][-max_seq_length + 1 :]
+            all_loss_mask[i] = [1 - ignore_start_end] + all_loss_mask[i][-max_seq_length + 1 :]
+            all_subtokens_mask[i] = [0] + all_subtokens_mask[i][-max_seq_length + 1 :]
+
+            if with_label:
+                all_slots[i] = [pad_label] + all_slots[i][-max_seq_length + 1 :]
+            too_long_count += 1
+
+        all_input_ids.append([tokenizer.tokens_to_ids(t) for t in subtokens])
+
+        if len(subtokens) < max_seq_length:
+            extra = max_seq_length - len(subtokens)
+            all_input_ids[i] = all_input_ids[i] + [0] * extra
+            all_loss_mask[i] = all_loss_mask[i] + [0] * extra
+            all_subtokens_mask[i] = all_subtokens_mask[i] + [0] * extra
+            all_input_mask[i] = all_input_mask[i] + [0] * extra
+
+            if with_label:
+                all_slots[i] = all_slots[i] + [pad_label] * extra
+
+        all_segment_ids.append([0] * max_seq_length)
+
+    logging.info(f'{too_long_count} are longer than {max_seq_length}')
+
+    logging.info("*** Some Examples of Processed Data***")
+    for i in range(min(len(all_input_ids), 5)):
+        logging.info("i: %s" % (i))
+        logging.info("subtokens: %s" % " ".join(list(map(str, all_subtokens[i]))))
+        logging.info("loss_mask: %s" % " ".join(list(map(str, all_loss_mask[i]))))
+        logging.info("input_mask: %s" % " ".join(list(map(str, all_input_mask[i]))))
+        logging.info("subtokens_mask: %s" % " ".join(list(map(str, all_subtokens_mask[i]))))
+        if with_label:
+            logging.info("slots_label: %s" % " ".join(list(map(str, all_slots[i]))))
+
+    return (all_input_ids, all_segment_ids, all_input_mask, all_loss_mask, all_subtokens_mask, all_slots)
+
+
+class IntentSlotClassificationDataset(Dataset):
+    """
```

Comment:
please add @experimental for now to the datasets, input/output types are missing. Please also add python3 hints

Reply:
Some inputs are missing from the doc string.

Reply:
Resolved

Reply:
Python3 hints are still misising.

Reply:
After adding the hints, you may drop the types in the doc strings

Reply:
Please remove the types in the doc section. Python 3 hints are still missing.